fix: save display settings in local storage

Chandru-HM opened the pull request:
# Description

This PR is for ticket https://github.com/awslabs/iot-app-kit/issues/2313 and display settings are now saved in local storage. 

## Type of change
- [x] Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

Display settings are now stored in local storage, verified saving the dashboard settings and refreshing the screen or switching to different dashboard.
[store-display-settings.webm](https://github.com/awslabs/iot-application/assets/142866907/3b8382cb-cb68-43aa-a610-65c43d1c30ad)


# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x]  I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

## Legal

This project is available under the [Apache 2.0 License](http://www.apache.org/licenses/LICENSE-2.0.html).

tracy-french reviewed (changes requested):
It looks good and is functional, but can we please write either unit, integration, or Playwright tests to ensure the functionality is maintained? Thank you! :)

tracy-french commented:
Thank you @Chandru-HM! Unfortunately, we now have a merge conflict on the test file as we updated `rc`. Could you please resolve that and we'll merge right away?

Chandru-HM commented:
> Thank you @Chandru-HM! Unfortunately, we now have a merge conflict on the test file as we updated `rc`. Could you please resolve that and we'll merge right away?

Updated. please have a look.

tracy-french commented:
Thank you @Chandru-HM!
